### PR TITLE
Add handling of negative returned values from historical balance

### PR DESCRIPTION
**Pull request by spiderjako**
There are certain cases, in which the returned values are unexpectedly negative and we want to avoid that.
https://api.santiment.net/graphiql?variables=%7B%7D&query=%7B%0A%20%20assetsHeldByAddress(address%3A%20%220x0000000000000000000000000000000000000000%22)%20%7B%0A%20%20%20%20balance%0A%20%20%20%20slug%0A%20%20%7D%0A%7D%0A

### CODE REVIEW

**Comment by IvanIvanoff** [comment]
Just a note: The negative balances are not a bug, but rather a consequence of our implementation of the balance calculation. Many contracts emit a transfer event when minting new tokens by setting 0x00...000 as the from address, hence 0x00...000 is "sending" tokens it does not have which leads to "negative" balance

File: test/sanbase_web/graphql/clickhouse/historical_balance/assets_held_by_address_api_test.exs [hunk 1]
```diff
@@ -44,7 +44,6 @@ defmodule SanbaseWeb.Graphql.Clickhouse.AssetsHeldByAdderssApiTest do
                "data" => %{
                  "assetsHeldByAddress" => [
                    %{"balance" => 1.0e3, "slug" => context.eth_project.coinmarketcap_id},
-                   %{"balance" => 100.0, "slug" => context.p1.coinmarketcap_id},
```
Comment (IvanIvanoff) [inline test/sanbase_web/graphql/clickhouse/historical_balance/assets_held_by_address_api_test.exs]: can we have a test that is explicitly named "negative balances are discarded" :?

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex [hunk 2]
```diff
@@ -324,6 +324,11 @@ defmodule SanbaseWeb.Graphql.Resolvers.ClickhouseResolver do
     HistoricalBalance.assets_held_by_address(address)
     |> case do
       {:ok, result} ->
+        result =
+          result
+          |> Enum.reject(fn row -> Map.get(row, :balance) < 0 end)
+          |> Enum.to_list()
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: The functions in the `Enum` module already return lists, no point in calling `to_list` here

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex [hunk 3]
```diff
@@ -324,6 +324,11 @@ defmodule SanbaseWeb.Graphql.Resolvers.ClickhouseResolver do
     HistoricalBalance.assets_held_by_address(address)
     |> case do
       {:ok, result} ->
+        result =
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: You can add a comment why we do this - check my note in the other comment describing the reason

File: test/sanbase_web/graphql/clickhouse/historical_balance/assets_held_by_address_api_test.exs (hunk 1 above)
Comment (spiderjako) [reply test/sanbase_web/graphql/clickhouse/historical_balance/assets_held_by_address_api_test.exs]: Yes, that's a better idea

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex (hunk 2 above)
Comment (spiderjako) [reply lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: I was wondering that :D thanks, will be changed

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex (hunk 3 above)
Comment (spiderjako) [reply lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: Of course, it shall be done

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex [hunk 4]
```diff
@@ -324,6 +324,17 @@ defmodule SanbaseWeb.Graphql.Resolvers.ClickhouseResolver do
     HistoricalBalance.assets_held_by_address(address)
     |> case do
       {:ok, result} ->
+        """
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: This is not a comment but a string declaration. For comments you should prepend `#` to every line

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex [hunk 5]
```diff
@@ -324,6 +324,17 @@ defmodule SanbaseWeb.Graphql.Resolvers.ClickhouseResolver do
...
+        We do this, because many contracts emit a transfer
+        event when minting new tokens by setting 0x00...000
+        as the from address, hence 0x00...000 is "sending"
+        tokens it does not have which leads to "negative" balance
+        """
+
+        result =
+          result
+          |> Enum.reject(fn row -> Map.get(row, :balance) < 0 end)
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]:
```suggestion
          |> Enum.reject(fn %{balance: balance} -> balance < 0 end)
```

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex (hunk 4 above)
Comment (spiderjako) [reply lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: Oh, that's right, thanks

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex (hunk 5 above)
Comment (spiderjako) [reply lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: That's a better idea really, thanks, it should be changed now

**Pull request merged**